java/core/src/java/org/apache/orc/impl/WriterImpl.java

```diff
@@ -300,7 +302,7 @@ private boolean checkMemory() throws IOException {
         LOG.debug("ORC writer " + physicalWriter + " size = " + size +
                       " limit = " + memoryLimit);
       }
-      if (size > memoryLimit) {
+      if (size > memoryLimit || rowsInStripe> stripeRowCount) {
```

Comment:
The frequency of this check is influenced by `OrcConf.ROWS_BETWEEN_CHECKS`, we should compare the value of `stripeRowCount` with `OrcConf.ROWS_BETWEEN_CHECKS` before using `stripeRowCount`.

Reply:
@guiyanakuang I use orc.stripe.row.count instead if rows_between_checks larger than orc.stripe.row.count.